src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1926,4 +1942,27 @@ public static String getVersionName() {
         // value
         return "1.1.12";
     }
+
+    /**
+     * A {@link Future}  whose result can be set by a {@link #set(Object)} or {@link #setException(Throwable)}
+     */
+    final static class SettableFuture<V> extends FutureTask<V> {
```

Comment:
After re-reading you original PR comment I think it get it a bit better.  However, I'm not sure I understand why this change is necessary.  Why is it a problem that the caller needs to call the blocking get() function?  Can't they do that after they are certain the operation has completed and therefore ensure that no blocking will actually occur?  I'm not saying that it is the best interface, just that if it is possible to make it work then I don't see any reason to change it at this time.